backend/src/api/routes/enhanced_guide.py

```
@@ -235,15 +235,18 @@ async def generate_enhanced_guide(
-        # save_result = await database_service.save_enhanced_guide_data(trip_id, guide)
-        # if not save_result.success:
-        #     logger.error(f"Failed to save guide: {save_result.error}")
-        #     raise HTTPException(status_code=500, detail=f"Failed to save guide: {save_result.error}")
-        # logger.info(f"Enhanced guide saved successfully for trip {trip_id}")
+        # Save guide to database
+        logger.info("Saving enhanced guide data to database")
+        try:
+            save_result = await database_service.save_enhanced_guide_data(trip_id, guide)
+            if not save_result.success:
+                logger.error(f"Failed to save guide: {save_result.error}")
+                raise HTTPException(status_code=500, detail=f"Failed to save guide: {save_result.error}")
```

Ruff (0.13.1)

Abstract `raise` to an inner function

(TRY301)